src/server/pps/server/api_server.go

```diff
@@ -742,6 +742,21 @@ func (a *apiServer) CreatePipeline(ctx context.Context, request *ppsclient.Creat
 		return nil, err
 	}
 
+	existingPipelineInfo, err := persistClient.GetPipelineInfo(ctx, request.Pipeline)
```

Comment:
@jdoliner pointed out a way to simplify this code a bunch. Just moved the repo creation to after creating the `PipelineInfo` ... now the race is prevented and the descriptive error is present before the repo is created. I'll just make sure to destroy the `PipelineInfo` if the repo creation fails.